EIPS/eip-2327.md

```diff
@@ -0,0 +1,49 @@
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+While computing the valid JUMPDESTs of a contract, halt analysis if `BEGINDATA` is encountered. A `JUMP` to a value higher than the `pc` value of `BEGINDATA` should throw with a `BAD_JUMP_DESTINATION` error.
+If `BEGINDATA` is encountered during contract execution, it has the same semantics as `STOP`. It uses 0 gas.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+The byte `0xb6` was chosen to align with [EIP-615](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-615.md). 
+The choice to STOP if `BEGINDATA` is encountered is somewhat arbitrary. An alternative would be to `THROW`.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+The proposal will not change any existing contracts unless their current behaviour relies upon the usage of unused opcodes.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+Test cases should include:
+1) A contract which jumps to a destination `X`, where `X` has a pc value higher than the `BEGINDATA` opcode, and the byte at `X` is `0x5b`. This should fail with a `BAD_JUMPDEST`.
```

Comment:
Please use BAD_JUMPDEST or BAD_JUMP_DESTINATION consistently.

Reply:
Albeit there is no such error clearly defined in EVM semantics, I'd still opt to keep that in the EIP (for now).

Reply:
I am referring to the status codes as defined by EVMC https://evmc.ethereum.org/group__EVMC.html#ga4c0be97f333c050ff45321fcaa34d920. Even if they haven't been universally accepted, they do provide the most systematic treatment of exceptions I'm aware of.

Reply:
I'm aware of them and glad to see them utilised, they haven't been referenced yet (apart from #2003).